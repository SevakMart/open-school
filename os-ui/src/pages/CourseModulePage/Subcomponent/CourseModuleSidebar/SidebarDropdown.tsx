import { t } from 'i18next';
import { COURSE_CATEGORIES } from '../../../../constants/CourseModuleCategories';
import styles from './Sidebar.module.scss';

interface Props {
handleChangeValue: (a:string) => void,
  }

const SidebarDropdown = ({ handleChangeValue }: Props) => (
  <>
    {
		COURSE_CATEGORIES.map((button: { value: string, id: string, title: string }) => (
  <label key={button.id} className={styles.Checkbox_container}>
    {t(`${button.title}`)}
    <input type="checkbox" onChange={() => handleChangeValue(button.value)} />
    <span className={styles.Checkbox_checkmark} />
  </label>
		))
	  }
  </>
);

export default SidebarDropdown;
