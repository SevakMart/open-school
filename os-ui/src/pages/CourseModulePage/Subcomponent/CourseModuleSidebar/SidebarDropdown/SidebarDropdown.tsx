import { t } from 'i18next';
import styles from '../Sidebar.module.scss';

interface Props {
	value:string,
  handleChangeValue: (a: string) => void,
  modules: {
    link?: any,
    title: string,
    description: string,
    moduleItemSet: { [index: string]: string }[]
  }[];
}

const SidebarDropdown = ({ handleChangeValue, value, modules }: Props) => (
  <>
    {
        modules.map((module) => (
          <label key={module.title} className={styles.Checkbox_container}>
            {t(`${module.title}`)}
            <input
              type="radio"
              name="category"
              value={module.title}
              checked={value === module.title}
              onChange={() => handleChangeValue(module.title)}
            />
            <span className={styles.Checkbox_checkmark} />
          </label>
        ))
      }
  </>
);

export default SidebarDropdown;
