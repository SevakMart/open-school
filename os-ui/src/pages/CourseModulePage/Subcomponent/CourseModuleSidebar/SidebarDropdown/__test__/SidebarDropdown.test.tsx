import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import SidebarDropdown from '../SidebarDropdown';

describe('SidebarDropdown component', () => {
  const handleChangeValue = jest.fn();
  const modules = [
    {
      title: 'Title 1',
      description: 'Description 1',
      moduleItemSet: [],
    },
    {
      title: 'Title 2',
      description: 'Description 2',
      moduleItemSet: [],
    },
  ];

  test('should call handleChangeValue with correct value when radio button is clicked', () => {
    const { container, asFragment } = render(
      <SidebarDropdown
        value={modules[0].title}
        handleChangeValue={handleChangeValue}
        modules={modules}
      />,
    );

    const radioBtn1 = container.querySelector('input[value="Title 1"]') as HTMLInputElement;
    const radioBtn2 = container.querySelector('input[value="Title 2"]') as HTMLInputElement;

    expect(radioBtn1.checked).toBe(true);

    fireEvent.click(radioBtn2);
    expect(handleChangeValue).toHaveBeenCalledWith('Title 2');

    expect(asFragment()).toMatchSnapshot();
  });
});
