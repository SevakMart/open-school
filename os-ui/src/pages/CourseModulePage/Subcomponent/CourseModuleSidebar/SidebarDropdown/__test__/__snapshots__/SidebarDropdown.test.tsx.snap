// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarDropdown component should call handleChangeValue with correct value when radio button is clicked 1`] = `
<DocumentFragment>
  <label
    class="Checkbox_container"
  >
    <input
      checked=""
      name="category"
      type="radio"
      value="Title 1"
    />
    <span
      class="Checkbox_checkmark"
    />
  </label>
  <label
    class="Checkbox_container"
  >
    <input
      name="category"
      type="radio"
      value="Title 2"
    />
    <span
      class="Checkbox_checkmark"
    />
  </label>
</DocumentFragment>
`;
